front_end/helper.py

```diff
@@ -110,15 +110,10 @@ def exec_code(settings_dict, code, exercise_basics, exercise_details, request=No
     response = requests.post(f"http://127.0.0.1:{middle_layer_port}/exec/", json.dumps(data_dict), timeout=timeout)
 
     response_dict = json.loads(response.content)
-    return response_dict["text_output"], response_dict["image_output"]
-
-def check_exercise_output(exercise_details, actual_text, actual_image):
-    if exercise_details["back_end"] == "any_response" and (len(actual_text) > 0 or len(actual_image) > 0):
-        return "", True
-
-    if exercise_details["output_type"] == "txt":
-        expected_text = exercise_details["expected_text_output"]
+    return response_dict["text_output"], response_dict["image_output"], json.loads(response_dict["tests"])
```

Comment:
Shouldn't there be a different text_output and image_output for each test? What does response_dict["tests"] contain? I assume this is different from the tests in exercise_details? Maybe rename the response key here so that it is more intuitive what is included in it and make a comment that clarifies it.